### PR TITLE
Add test for #32108 (error with groupby on series with period index)

### LINKED ISSUE #32108: Series with NAMED period index raise error on groupby index.month (pandas 1.0 specific)
*edit from @TomAugspurger*: this is fixed on master, but the example below needs to be added as a unit test. The test can probably go in `groupby/test_groupby.py`.

#### Description

With the pandas 1.0.1 (full version with dependencies at the end), series with NAMED period index  raise error on groupby index.month

There is no error if the index is not named.

There was no error wit pandas 0.25.3

#### Code Sample
```python
import pandas as pd

index = pd.period_range(start='2018-01', periods=24, freq='M')
periodSerie = pd.Series(range(24),index=index)
periodSerie.index.name = 'Month'
periodSerie.groupby(periodSerie.index.month).sum()
```

#### Error
It seems to me that pandas tries to interpret the index name as if it were part of the index itself.

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/.virtualenvs/dev/lib/python3.8/site-packages/pandas/core/indexes/base.py in get_value(self, series, key)
   4410             try:
-> 4411                 return libindex.get_value_at(s, key)
   4412             except IndexError:

pandas/_libs/index.pyx in pandas._libs.index.get_value_at()

pandas/_libs/index.pyx in pandas._libs.index.get_value_at()

pandas/_libs/util.pxd in pandas._libs.util.get_value_at()

pandas/_libs/util.pxd in pandas._libs.util.validate_indexer()

TypeError: 'str' object cannot be interpreted as an integer

During handling of the above exception, another exception occurred:

KeyError                                  Traceback (most recent call last)
~/.virtualenvs/dev/lib/python3.8/site-packages/pandas/core/indexes/period.py in get_value(self, series, key)
    516         try:
--> 517             value = super().get_value(s, key)
    518         except (KeyError, IndexError):

~/.virtualenvs/dev/lib/python3.8/site-packages/pandas/core/indexes/base.py in get_value(self, series, key)
   4418                 else:
-> 4419                     raise e1
   4420             except Exception:

~/.virtualenvs/dev/lib/python3.8/site-packages/pandas/core/indexes/base.py in get_value(self, series, key)
   4404         try:
-> 4405             return self._engine.get_value(s, k, tz=getattr(series.dtype, "tz", None))
   4406         except KeyError as e1:

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_value()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_value()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/index_class_helper.pxi in pandas._libs.index.Int64Engine._check_type()

KeyError: 'Month'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
pandas/_libs/tslibs/parsing.pyx in pandas._libs.tslibs.parsing.parse_datetime_string_with_reso()

pandas/_libs/tslibs/parsing.pyx in pandas._libs.tslibs.parsing.dateutil_parse()

ValueError: Unknown datetime string format, unable to parse: Month

During handling of the above exception, another exception occurred:

DateParseError                            Traceback (most recent call last)
<ipython-input-6-a3e948d22d88> in <module>
      5 periodSerie = pd.Series(range(24),index=index)
      6 periodSerie.index.name = 'Month'
----> 7 periodSerie.groupby(periodSerie.index.month).sum()

~/.virtualenvs/dev/lib/python3.8/site-packages/pandas/core/series.py in groupby(self, by, axis, level, as_index, sort, group_keys, squeeze, observed)
   1676         axis = self._get_axis_number(axis)
   1677 
-> 1678         return groupby_generic.SeriesGroupBy(
   1679             obj=self,
   1680             keys=by,

~/.virtualenvs/dev/lib/python3.8/site-packages/pandas/core/groupby/groupby.py in __init__(self, obj, keys, axis, level, grouper, exclusions, selection, as_index, sort, group_keys, squeeze, observed, mutated)
    400             from pandas.core.groupby.grouper import get_grouper
    401 
--> 402             grouper, exclusions, obj = get_grouper(
    403                 obj,
    404                 keys,

~/.virtualenvs/dev/lib/python3.8/site-packages/pandas/core/groupby/grouper.py in get_grouper(obj, key, axis, level, sort, observed, mutated, validate)
    583     for i, (gpr, level) in enumerate(zip(keys, levels)):
    584 
--> 585         if is_in_obj(gpr):  # df.groupby(df['name'])
    586             in_axis, name = True, gpr.name
    587             exclusions.append(name)

~/.virtualenvs/dev/lib/python3.8/site-packages/pandas/core/groupby/grouper.py in is_in_obj(gpr)
    577             return False
    578         try:
--> 579             return gpr is obj[gpr.name]
    580         except (KeyError, IndexError):
    581             return False

~/.virtualenvs/dev/lib/python3.8/site-packages/pandas/core/series.py in __getitem__(self, key)
    869         key = com.apply_if_callable(key, self)
    870         try:
--> 871             result = self.index.get_value(self, key)
    872 
    873             if not is_scalar(result):

~/.virtualenvs/dev/lib/python3.8/site-packages/pandas/core/indexes/period.py in get_value(self, series, key)
    518         except (KeyError, IndexError):
    519             if isinstance(key, str):
--> 520                 asdt, parsed, reso = parse_time_string(key, self.freq)
    521                 grp = resolution.Resolution.get_freq_group(reso)
    522                 freqn = resolution.get_freq_group(self.freq)

pandas/_libs/tslibs/parsing.pyx in pandas._libs.tslibs.parsing.parse_time_string()

pandas/_libs/tslibs/parsing.pyx in pandas._libs.tslibs.parsing.parse_datetime_string_with_reso()

DateParseError: Unknown datetime string format, unable to parse: Month
```

#### Expected Output
With pandas 0.25.3, the following expected output is produced :

```python
Month
1     12
2     14
3     16
4     18
5     20
6     22
7     24
8     26
9     28
10    30
11    32
12    34
dtype: int64
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.8.1.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.18-1-MANJARO
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : fr_FR.UTF-8
LOCALE           : fr_FR.UTF-8

pandas           : 1.0.1
numpy            : 1.18.0
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 44.0.0
Cython           : 0.29.15
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.3
IPython          : 7.11.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.2
numexpr          : None
odfpy            : None
openpyxl         : 3.0.2
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : None
numba            : None
</details>

### PR DESCRIPTION
**Pull request by daxid**
- [ ] closes #32108

### CODE REVIEW

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -2057,3 +2057,15 @@ def test_groups_repr_truncates(max_seq_items, expected):
 
         result = df.groupby(np.array(df.a)).groups.__repr__()
         assert result == expected
+
+
+def test_groupby_period_index():
```
Comment (MarcoGorelli) [inline pandas/tests/groupby/test_groupby.py]: Add the issue number here:

```
def test_groupby_period_index():
    # GH 32108
```

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -2057,3 +2057,15 @@ def test_groups_repr_truncates(max_seq_items, expected):
...
+    periods = 2
+    index = period_range(start='2018-01', periods=periods, freq='M')
```
Comment (MarcoGorelli) [inline pandas/tests/groupby/test_groupby.py]: Looks like you haven't run `black`, and perhaps not `flake8` either. See [code standards](https://pandas.pydata.org/docs/development/contributing.html#code-standards)

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @daxid !

**Comment by dbeniamine** [comment]
I've done the corrections, all tests are passing now, sorry for the delay

File: pandas/tests/groupby/test_groupby.py [hunk 3]
```diff
@@ -2057,3 +2057,16 @@ def test_groups_repr_truncates(max_seq_items, expected):
 
         result = df.groupby(np.array(df.a)).groups.__repr__()
         assert result == expected
+
+
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py]: can you move this under  test_grouper_index_types (same file). ping on green.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/test_groupby.py (hunk 3 above)
Comment (dbeniamine) [reply pandas/tests/groupby/test_groupby.py]: The function `test_grouper_index_types` is in the file `pandas/tests/groupby/test_grouping.py` our test is in `pandas/tests/groupby/test_groupby.py`, do you want us to move our test to `test_grouping.py` ?

Comment (jreback) [reply pandas/tests/groupby/test_groupby.py]: yes pls move this

Comment (dbeniamine) [reply pandas/tests/groupby/test_groupby.py]: done

File: pandas/tests/groupby/test_grouping.py [hunk 4]
```diff
@@ -149,6 +149,18 @@ def test_grouper_index_types(self):
             df.index = list(reversed(df.index.tolist()))
             df.groupby(list("abcde")).apply(lambda x: x)
 
+    def test_grouper_period_index(self):
+        # GH 32108
+        periods = 2
+        index = pd.period_range(start="2018-01", periods=periods, freq="M")
+        periodSerie = pd.Series(range(periods), index=index)
```
Comment (MarcoGorelli) [inline pandas/tests/groupby/test_grouping.py:156]: naming this `periodSerie` doesn't conform to PEP8 - sorry for not bringing this up sooner, I thought flake8 would've caught it. seems there's an open issue to add extra plugins though (#20588)

Function and Variable Names
Function names should be lowercase, with words separated by underscores as necessary to improve readability.

> Variable names follow the same convention as function names.
> 
> mixedCase is allowed only in contexts where that's already the prevailing style (e.g. threading.py), to retain backwards compatibility.

https://www.python.org/dev/peps/pep-0008/

Comment (dbeniamine) [reply pandas/tests/groupby/test_grouping.py]: Done, I hope it's all good now :)

**Comment by simonjayhawkins** [comment]
@daxid thanks for working on this. seems to have been lost in the shuffle. can you merge master to trigger ci.

**Comment by dbeniamine** [comment]
> @daxid thanks for working on this. seems to have been lost in the shuffle. can you merge master to trigger ci.

done

**Comment by jreback** [comment]
can you merge master

**Comment by dbeniamine** [comment]
> can you merge master

done

File: pandas/tests/groupby/test_grouping.py [hunk 5]
```diff
@@ -149,6 +149,18 @@ def test_grouper_index_types(self):
...
 
+    def test_grouper_period_index(self):
+        # GH 32108
+        periods = 2
+        index = pd.period_range(start="2018-01", periods=periods, freq="M")
+        period_serie = pd.Series(range(periods), index=index)
+        period_serie.index.name = "Month"
+        result = period_serie.groupby(period_serie.index.month).sum()
+
+        expected = pd.Series(range(0, periods), index=range(1, periods + 1))
```
Comment (jreback) [inline pandas/tests/groupby/test_grouping.py:160]: you can construct the `Index(...., name=...)`

File: pandas/tests/groupby/test_grouping.py [hunk 6]
```diff
@@ -149,6 +149,18 @@ def test_grouper_index_types(self):
             df.index = list(reversed(df.index.tolist()))
             df.groupby(list("abcde")).apply(lambda x: x)
 
+    def test_grouper_period_index(self):
+        # GH 32108
+        periods = 2
+        index = pd.period_range(start="2018-01", periods=periods, freq="M")
```
Comment (jreback) [inline pandas/tests/groupby/test_grouping.py:155]: pass `name=`

File: pandas/tests/groupby/test_grouping.py [hunk 7]
```diff
@@ -149,6 +149,18 @@ def test_grouper_index_types(self):
...
+        period_serie = pd.Series(range(periods), index=index)
```
Comment (jreback) [inline pandas/tests/groupby/test_grouping.py:156]: just name this period_series

File: pandas/tests/groupby/test_grouping.py [hunk 8]
```diff
@@ -149,6 +149,18 @@ def test_grouper_index_types(self):
             df.index = list(reversed(df.index.tolist()))
             df.groupby(list("abcde")).apply(lambda x: x)
 
+    def test_grouper_period_index(self):
+        # GH 32108
```
Comment (jreback) [inline pandas/tests/groupby/test_grouping.py:153]: move this to pandas/tests/groupby/test_timegrouper.py

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by dbeniamine** [comment]
Done once again

File: pandas/tests/groupby/test_timegrouper.py [hunk 9]
```diff
@@ -769,3 +769,18 @@ def test_scalar_call_versus_list_call(self):
...
+        periods = 2
+        index = pd.period_range(
+            start="2018-01", periods=periods, freq="M", name="Month"
+        )
+        period_serie = pd.Series(range(periods), index=index)
+        result = period_serie.groupby(period_serie.index.month).sum()
+
+        expected = pd.Series(
+            range(0, periods),
+            index=Index(range(1, periods + 1), name=period_serie.index.name),
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/test_timegrouper.py:784]: maybe have name as index.name or just be explicit and use name="Month"

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @daxid generally lgtm can you address https://github.com/pandas-dev/pandas/pull/33105#discussion_r440134823

**Comment by dbeniamine** [comment]
Is that version good for you ?

**Comment by simonjayhawkins** [comment]
it appears that https://github.com/pandas-dev/pandas/pull/33105#discussion_r440134823 has not been addressed period_serie -> period_series otherwise lgtm. Thanks @daxid

**Comment by dbeniamine** [comment]
> it appears that [#33105 (comment)](https://github.com/pandas-dev/pandas/pull/33105#discussion_r440134823) has not been addressed period_serie -> period_series otherwise lgtm. Thanks @daxid

I am not sure to understand you request, do you want us to add a `name=` parameter  in the `period_serie` construction or do you want us to change the name `period_serie` to something else, maybe `month_serie` ?

File: pandas/tests/groupby/test_grouping.py (hunk 7 above)
Comment (jreback) [reply pandas/tests/groupby/test_grouping.py]: right can you construct the initial index like

```
> index = pd.period_range(start="2018-01", periods=periods, freq="M")
index = pd.period_range(start="2018-01", periods=periods, freq="M", name='foo')
```

Comment (dbeniamine) [reply pandas/tests/groupby/test_grouping.py]: The line on the current version is already:

```python
index = pd.period_range(
            start="2018-01", periods=periods, freq="M", name="Month"
        )
```

**Comment by MarcoGorelli** [comment]
> > it appears that [#33105 (comment)](https://github.com/pandas-dev/pandas/pull/33105#discussion_r440134823) has not been addressed period_serie -> period_series otherwise lgtm. Thanks @daxid
> 
> I am not sure to understand you request, do you want us to add a `name=` parameter in the `period_serie` construction or do you want us to change the name `period_serie` to something else, maybe `month_serie` ?

Pretty sure the request is to rename `period_serie` to `period_series`

**Comment by dbeniamine** [comment]
> Pretty sure the request is to rename `period_serie` to `period_series`

Thanks did now saw the `s` so I wasn't understing this request, should be good now

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @daxid

**Pull request merged**